File: mediapicker/src/main/java/iam/thevoid/mediapicker/util/Editor.java
package iam.thevoid.mediapicker.util;

import java.util.Date;

/**
 * Created by iam on 22/08/2017.
 */

public final class Editor {
    private Editor() {
    }

    public static String currentDateFilename(String prefix, String ext) {
        if (ext == null) {
            ext = "";
        }

        if (prefix == null) {
            prefix = "";
        }

        return String.format("%s%s%s",
                (prefix.length() == 0 ? "" : prefix + "_"),
                DateManager.formatDateToString(
                        "yyyy_MM_dd_HH_mm_ss", new Date(DateManager.getTime())),
                ext
        );
    }

}

File: mediapicker/src/main/java/iam/thevoid/mediapicker/util/Editor.kt
@file:JvmName("Editor")

package iam.thevoid.mediapicker.util

import iam.thevoid.e.orElse

/**
 * Created by iam on 22/08/2017.
 */

fun currentDateFilename(extension: String): String =
        "${filenamePrefix(extension).orElse("file")}_${formatDateToString("yyyy_MM_dd_HH_mm_ss")}$extension"

private fun filenamePrefix(extension: String): String =
        when {
            FileUtil.isVideoExt(extension) -> "video"
            FileUtil.isGifExt(extension) -> "anim"
            else -> "image"
        }
